pkg/grpc/requestinfo/headers_multimap.go

```diff
@@ -0,0 +1,19 @@
+package requestinfo
+
+import (
+	"net/http"
+)
+
+// HeadersMultiMap is an interface for http.Header and for metadata.MD types.
+// The former will require a wrapper to implement this interface.
+type HeadersMultiMap interface {
+	Get(key string) []string
+}
+
+// AsHeadersMultiMap adds the HeadersMultiMap implementation to http.Header.
+type AsHeadersMultiMap http.Header
```

Comment:
I'll think how to better name this type. It has to be different from the interface name.

Reply:
It would look fine if it were a function, I guess. But it is actually used as a function:
`requestinfo.AsHeadersMultiMap(req.Header)`.
I wouldn't add a function just to justify the name.

Reply:
What about just `Headers`, so it will read `var m HeadersMultiMap = requestinfo.Headers(req.Header)`?

Reply:
Sounds good to me